src/coreclr/scripts/superpmi.py

```diff
@@ -3897,28 +3898,75 @@ def download_mch_from_azure(coreclr_args, target_dir):
         list containing the local path of files downloaded
     """
 
+    blob_url_prefix = "{}/{}/".format(az_blob_storage_superpmi_container_uri, az_collections_root_folder)
     blob_filter_string =  "{}/{}/{}/".format(coreclr_args.jit_ee_version, coreclr_args.target_os, coreclr_args.mch_arch).lower()
 
-    # Determine if a URL in Azure Storage should be allowed. The path looks like:
-    #   jit-ee-guid/Linux/x64/Linux.x64.Checked.frameworks.mch.zip
-    # Filter to just the current jit-ee-guid, OS, and architecture.
-    # Include both MCH and MCT files as well as the CLR JIT dll (processed below).
-    # If there are filters, only download those matching files.
-    def filter_superpmi_collections(path):
-        path = path.lower()
-        return path.startswith(blob_filter_string) and ((coreclr_args.filter is None) or any((filter_item.lower() in path) for filter_item in coreclr_args.filter))
+    path_var = os.environ.get("PATH")
+    azcopy_exe = "azcopy.exe" if platform.system() == "Windows" else "azcopy"
+    azcopy_path = find_file(azcopy_exe, path_var.split(os.pathsep)) if path_var is not None else None
+
+    if azcopy_path is None or authenticate_using_azure:
```

Comment:
I don't see we ever set this to `True`. Do you know? Also, i think this should be checking for `... or not authenticate_using_azure`

Reply:
I don't know, I guess it just allows you to change it in the source if you have an internal storage account that needs auth.

> I don't see we ever set this to `True`. Do you know? Also, i think this should be checking for `... or not authenticate_using_azure`

We need to fall back to the old behavior when it is true because the `azcopy` path does not do the necessary authentication. So it should be right as-is.